### PR TITLE
Support reloading and improve local storage

### PR DESCRIPTION
**Pull request by charles-tan**
Added a timestamp field to local storage so we know when the TC string was last fetched. Then added this to the popup UI.

Also added a function to prune the local tab storage when the cookie glasses popup is activated. It checks if there are over 200 items in local storage map and if there are, remove any that are related to uCookie and are older than 12 hours.

Added support for reloading a page. Now when pages are loaded, a connection between background script and uCookie script is attempted.

Looks to address https://github.com/katie-ta/Cookie-Glasses/issues/18 and https://github.com/katie-ta/Cookie-Glasses/issues/19

### CODE REVIEW

File: src/content_scripts/uCookie.js [hunk 1]
```diff
@@ -135,3 +136,5 @@ api.runtime.onConnect.addListener((port) => {
   backgroundPort = port;
   backgroundPort.onMessage.addListener(handleMessage);
 });
+
+// on page load
```
Comment (katie-ta) [inline src/content_scripts/uCookie.js:140]: can we delete this?

File: src/js/background.js [hunk 2]
```diff
@@ -94,13 +98,15 @@ function setUpConnection() {
 }
 
 api.tabs.onActivated.addListener(() => {
-  if (contentScriptPort) {
-    console.log('disconnecting port', contentScriptPort);
-    contentScriptPort.disconnect();
-  }
-
```
Comment (katie-ta) [inline src/js/background.js:101]: why should we move this?

File: src/popup/popup.js [hunk 3]
```diff
@@ -91,50 +94,119 @@ function showPurposes(purposeConsents) {
...
+  document.getElementById('last_fetched').textContent = formatIntlDate(lastFetched);
 }
 
-function handleTCData(data) {
+function handleTCData(data, timestamp) {
   showCmp(data.cmpId_);
   showNumVendors(data.vendorConsents);
   showPurposes(data.purposeConsents);
-  showTimestamps(data.created, data.lastUpdated);
+  showTimestamps(data.created, data.lastUpdated, timestamp);
```
Comment (katie-ta) [inline src/popup/popup.js:130]: can we rename `timestamp` to better reflect what timestamp it represents? i.e. `loadedTCDataTimestamp` or something

File: src/popup/popup.js [hunk 4]
```diff
@@ -91,50 +94,119 @@ function showPurposes(purposeConsents) {
...
 function getActiveTabStorage() {
-  api.tabs.query({ active: true, currentWindow: true }, (tabs) => {
-    const activeTabId = tabs[0].id;
-    console.log('active tab id', tabs[0].id);
-
-    api.storage.local.get([String(activeTabId)], (result) => {
-      const data = result[activeTabId];
-      console.log('data from storage', data);
+  let count = 0;
+  const retries = 3;
```
Comment (katie-ta) [inline src/popup/popup.js:135]: can we move this to the top of the file and rename in uppercase? i.e. `const MAX_RETRIES = 3`

File: src/popup/popup.js [hunk 5]
```diff
@@ -91,50 +94,119 @@ function showPurposes(purposeConsents) {
...
+      const activeTabId = tabs[0].id;
+      console.log('active tab id', tabs[0].id);
+
+      api.storage.local.get([String(activeTabId)], (result) => {
+        const data = result[activeTabId];
+        console.log('data from storage', data);
+        if (data === undefined) {
+          if (count <= retries) {
+            document.getElementById('nothing_found').classList.add('hidden');
+            document.getElementById('error_fetching_retry').classList.remove('hidden');
```
Comment (katie-ta) [inline src/popup/popup.js:148]: can we use the `hideElement` and `showHiddenElement` helpers instead?

**Review by katie-ta — APPROVED** [review]
a few nits, but logic lgtm. thank you for handling this!

File: src/js/background.js (hunk 2 above)
Comment (charles-tan) [reply src/js/background.js:101]: We probably want to make sure we disconnect the existing port before connecting to a new tab. Moving it to the function that we actually connect the tab will ensure this.

**Pull request merged**